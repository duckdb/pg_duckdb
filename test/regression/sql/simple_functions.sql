CREATE FUNCTION test_escape_uri(
    input TEXT
)
RETURNS TEXT
SET search_path = pg_catalog, pg_temp
LANGUAGE C AS 'pg_duckdb', 'pgduckdb_test_escape_uri';

SELECT test_escape_uri('https://duckdb.org');

SELECT test_escape_uri('https://duckdb.org/with space');

SELECT test_escape_uri('foo $ bar # baz / qux');

SELECT test_escape_uri('foo ðŸ˜€ bar # baz / qux');

SELECT test_escape_uri('Hannes MÃ¼hleisen');

SELECT test_escape_uri('Hannes Mâ–¡hleisen');

SELECT test_escape_uri('some 19 really $  - @ weird name ðŸ˜€ 84');
