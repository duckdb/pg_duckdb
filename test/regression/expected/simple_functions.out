CREATE FUNCTION test_escape_uri(
    input TEXT
)
RETURNS TEXT
SET search_path = pg_catalog, pg_temp
LANGUAGE C AS 'pg_duckdb', 'pgduckdb_test_escape_uri';
SELECT test_escape_uri('https://duckdb.org');
     test_escape_uri      
--------------------------
 https%3A%2F%2Fduckdb.org
(1 row)

SELECT test_escape_uri('https://duckdb.org/with space');
             test_escape_uri             
-----------------------------------------
 https%3A%2F%2Fduckdb.org%2Fwith%20space
(1 row)

SELECT test_escape_uri('foo $ bar # baz / qux');
             test_escape_uri             
-----------------------------------------
 foo%20%24%20bar%20%23%20baz%20%2F%20qux
(1 row)

